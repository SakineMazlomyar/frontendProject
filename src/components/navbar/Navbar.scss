/* Navbar style */
@import '../../sharedStyles/globalVariablesScss.scss';

.navbar {
    position: fixed;
    top: 0;
    width: $navbar-width;
    background-color: $navbar-background-color;
    height: $navbar-height;
    overflow: hidden;
    border:  2px solid $bordercolor-light;

    .navbarItemContainer {
        display: flex;
        flex-direction: row;
        justify-content: space-around;
        list-style: none;
    
        .navbarItem {
    
            .active {
                color: $color-blue;
                text-decoration: underline;
            }
            a {
                text-decoration: none;
                color: $navbar-text-color
            }
            
            &::first-letter{ text-transform: uppercase; }
        }
    }
}


// Small devices phones, tablets ( landscape 700px and down )
@media only screen and ( max-width: $min-tablet-width ) {
    .navbar {
        .navbarItemContainer {
         display: flex;
         flex-direction: column;
     
         .navbarItem {
             padding-bottom: $padding-smaller;
     
             .active {
                     font-size: $font-size-small;
                 }
             }
         }
    }
}
