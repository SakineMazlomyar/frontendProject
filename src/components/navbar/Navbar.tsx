import React from 'react'
import { NavLink } from 'react-router-dom'
import './Navbar.scss'

const Navbar: React.FC = () => {
    return ( <nav className="navbar" data-testid="navbar">
                <ul className="navbarItemContainer ">
                    <li className="navbarItem"><NavLink exact to="/" activeClassName="active">home</NavLink></li>
                    <li className="navbarItem"><NavLink exact to="/counter" activeClassName="active" id="counter-page">counter</NavLink></li>
                </ul>
            </nav>) 
}

export default Navbar