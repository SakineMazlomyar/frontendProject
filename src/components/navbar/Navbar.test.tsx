import Navbar from './Navbar'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json';

configure( { adapter: new Adapter() } ) 

describe('Navbar component', () => {

    test('shoud exist and match the snapshot', () => {
        const wrapper = shallow( <Navbar /> );
         expect( toJson( wrapper ) ).toMatchSnapshot();
    })
})
