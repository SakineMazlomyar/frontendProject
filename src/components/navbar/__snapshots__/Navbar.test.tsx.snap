// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component shoud exist and match the snapshot 1`] = `
<nav
  className="navbar"
  data-testid="navbar"
>
  <ul
    className="navbarItemContainer "
  >
    <li
      className="navbarItem"
    >
      <NavLink
        activeClassName="active"
        exact={true}
        to="/"
      >
        home
      </NavLink>
    </li>
    <li
      className="navbarItem"
    >
      <NavLink
        activeClassName="active"
        exact={true}
        id="counter-page"
        to="/counter"
      >
        counter
      </NavLink>
    </li>
  </ul>
</nav>
`;

exports[`Navbar component shoud exist and match the snapshot 1`] = `
<nav
  className="navbar"
  data-testid="navbar"
>
  <ul
    className="navbarItemContainer "
  >
    <li
      className="navbarItem"
    >
      <NavLink
        activeClassName="active"
        exact={true}
        to="/"
      >
        home
      </NavLink>
    </li>
    <li
      className="navbarItem"
    >
      <NavLink
        activeClassName="active"
        exact={true}
        id="counter-page"
        to="/counter"
      >
        counter
      </NavLink>
    </li>
  </ul>
</nav>
`;
